docs/adr/0009_plugin_registry.md

```diff
@@ -0,0 +1,306 @@
+all the available plugins and where to download them from:
+
+```yaml
+# https://plugins.ocm.software/index.yaml
+apiVersion: v1
+entries:
+  ecrplugin:
+    - name: ecrplugin
+      version: 0.27.0
+      description: AWS ECR repository plugin
+      home: https://github.com/ocm/ecrplugin
+      sources:
+        - https://github.com/ocm/ecrplugin
+      created: "2025-09-01T10:00:00Z"
+      digest: sha256:abc123...
+      downloads:
+        - os: linux
+          architecture: amd64
+          url: https://plugins.ocm.software/ecrplugin/0.27.0/ecrplugin-linux-amd64
+          sha256: ac3f340100668ff6e6c8f43284b6763893d652edb2333f9a8e97f21478c3e393
+        - os: darwin
+          architecture: amd64
+          url: https://plugins.ocm.software/ecrplugin/0.27.0/ecrplugin-darwin-amd64
+          sha256: f1764c6d84cdcae35886d02441de1cc81c7df02a3f5bb7710a1418bfe1b985c0
+        - os: windows
+          architecture: amd64
+          url: https://plugins.ocm.software/ecrplugin/0.27.0/ecrplugin-windows-amd64.exe
+          sha256: c81bf976fec0007476b738af769947e75544ca1661369e8d30403bab6553224e
+
+  helmInput:
+    - name: helmInput
+      version: 0.5.2
+      description: Helm input method plugin
+      home: https://github.com/ocm/helmInput
+      created: "2025-08-15T14:30:00Z"
+      digest: sha256:def456...
+      downloads:
+        - os: linux
+          architecture: amd64
+          url: https://plugins.ocm.software/helminput/0.5.2/helminput-linux-amd64
+          sha256: 112f227eb5310d4b9f85c0bf5ca0ad97464c0ea3f54c936bdecdebba01143c6a
+
+generated: "2025-09-24T10:00:00Z"
+```
+
+### Index Generation
+
+Registry maintainers update the index by:
+
+```bash
+# Generate index.yaml from plugin directory
+ocm plugin registry generate-index ./plugins/ > index.yaml
+
+# Upload to registry
+aws s3 cp index.yaml s3://plugins.ocm.software/
+aws s3 sync ./plugins/ s3://plugins.ocm.software/
+```
+
+### User Workflow for Index-based registries
+
+The user workflow largely remains the same as outlined in [User Workflow](#user-workflow). The underlying implementation
+changes, of course, to use the index instead of the registry component.
+
+## Pros and Cons
+
+### Component Version Registry Approach
+
+#### Pros
+
+It integrates with everything ocm already does; repositories, credentials, signatures, verification, distribution,
+download, upload, etc.
+It can include private registries and enterprise deployments, and it's consistent with how other OCM artifacts work.
+
+We can reuse a large portion of the existing `ocm download plugin` implementation, and existing workflows keep working.
+Component descriptors also give you metadata, labels, and provenance information.
+
+#### Cons
+
+It does require some ocm knowledge to host registries.
+Plugin publishers need to understand component versions, and setting up registries requires ocm tooling.
```

Comment:
is there a way we can simplify this in the future? should be described imo

Reply:
I don't know how honestly. It's a component version. It's our whole thing. How would you ever remove / simplify that fact?

Reply:
For example by offering helper CLI commands to package, build and publish a plugin. Just ideas

Reply:
I mean yes, but don't you want to understand what the hell you're pushing into your system? :D

Reply:
Yes, but I think its so many steps for build / package that helpers are inevitable at some point, even if we opinionate.